src/library/scala/runtime/LazyRef.scala

```diff
@@ -1,52 +1,157 @@
+/*                     __                                                      *\
+**     ________ ___   / /  ___     Scala API                                   **
+**    / __/ __// _ | / /  / _ |    (c) 2002-2016, LAMP/EPFL and Lightbend, Inc **
+**  __\ \/ /__/ __ |/ /__/ __ |    http://scala-lang.org/                      **
+** /____/\___/_/ |_/____/_/ | |                                                **
+**                          |/                                                 **
+\*                                                                             */
+
 package scala.runtime
 
-/* Classes used as holders for local lazy vals */
+/** Classes used as holders for lazy vals defined in methods. */
 
 class LazyRef[T] {
-  var value: T = _
-  @volatile var initialized: Boolean = false
+  @volatile private[this] var _initialized: Boolean = _
```

Comment:
Apologies, I guess what threw me of course is that this class (or rather these classes) are public, and non-final, classes which cannot be used properly without sprinkling of "magic"/macros. Are these intended to be possible to use *directly* by users?